2996_reflection/reflection.md

```diff
@@ -4232,38 +4242,42 @@ Modify paragraph 15 to disallow returning non-consteval-only pointers and refere
 
 or a prvalue core constant expression whose value satisfies the following constraints:
 
-* [#.#]{.pnum} if the value is an object of class type, each non-static data member of reference type refers to an [entity]{.rm} [object or function]{.addu} that is a permitted result of a constant expression, [and if the object or function is of consteval-only type, or is the subobject of such an object, then so is the value,]{.addu}
-* [#.#]{.pnum} if the value is an object of scalar type, it does not have an indeterminate or erroneous value ([basic.indet]{.sref}),
-* [#.#]{.pnum} if the value is of pointer type, [then:\ ]{.addu}
-  * [#.#.#]{.pnum} it is a pointer to an object of static storage duration[,]{.rm} [or]{.addu} a pointer past the end of such an object ([expr.add]{.sref}), [and if the complete object of that object is of consteval-only type then so is that pointer,]{.addu}
-  * [#.#.#]{.pnum} [it is]{.addu} a pointer to a non-immediate function, [and if that function is of consteval-only type then so is that pointer,]{.addu} or
-  * [#.#.#]{.pnum} [it is]{.addu} a null pointer value,
-* [#.#]{.pnum} if the value is of pointer-to-member-function type, it does not designate an immediate function, [and if that function is of consteval-only type then so is that value,]{.addu} and
-* [#.#]{.pnum} if the value is an object of class or array type, each subobject satisfies these constraints for the value.
+* [#.#]{.pnum} each constituent reference refers to an object or a non-immediate function,
+* [#.#]{.pnum} no constituent value of scalar type is an indeterminate value ([basic.indet]{.sref}),
+* [#.#]{.pnum} no constituent value of pointer type is a pointer to an immediate function or an invalid pointer value ([basic.compound]), [and]{.rm}
+* [#.#]{.pnum} no constituent value of pointer-to-member type designates an immediate function[.]{.rm}[, and]{.addu}
+
+::: addu
+* [#.#]{.pnum} unless the value is of consteval-only type, no constituent value of pointer or pointer-to-member type is
+  * a pointer to a function or member of consteval-only type, or
+  * a pointer to or past an object whose complete object has consteval-only type,
+
+  nor does any constituent reference refer to an object or function of consteval-only type.
```

Comment:
This is basically to reject type erasure? Like `struct X { void* p; };` where `p` points to an object of consteval-only type?

Reply:
Yep